tests/testthat/test-epi_slide.R

```diff
@@ -4,31 +4,55 @@ edf = dplyr::bind_rows(
   dplyr::tibble(geo_value = "ak", time_value = as.Date("2020-01-01") + 1:200, value=1:200),
   dplyr::tibble(geo_value = "al", time_value=as.Date("2020-01-01") + 1:5, value=-(1:5))
 ) %>%
-  as_epi_df()
+  as_epi_df() %>%
+  group_by(geo_value)
 
 f = function(x, ...) dplyr::tibble(value=mean(x$value), count=length(x$value))
 
 ## --- These cases generate the error: ---
+test_that("`after` must be defined as a non-zero integer if `before` is missing", {
+  expect_error(epi_slide(edf, f, after = 0L, ref_time_values=as.Date("2020-01-01")),
+               "`before` cannot be missing when `after` is set to 0.")
+})
+
+test_that("Warn user against having a blank `before`",{
+  expect_warning(epi_slide(edf, f, after = 1L, ref_time_values=as.Date("2020-01-01")+1L),
+                 regexp="`before` missing but `after` nonzero; `before` has been set to 0.")
+})
+
+test_that("Both `before` and `after` must be nonnegative integers",{
+  expect_error(epi_slide(edf, f, before = -1L, ref_time_values=as.Date("2020-01-01")+2L),
+               "`before` and `after` must be at least 0.")
+  expect_error(epi_slide(edf, f, before = 2L, after = -1L, ref_time_values=as.Date("2020-01-01")+2L),
+               "`before` and `after` must be at least 0.")
+  expect_error(epi_slide(edf, f, before = 0.5, ref_time_values=as.Date("2020-01-01")+2L),
+               "`before` and `after` must be integers.")
+  expect_error(epi_slide(edf, f, before = 1, after = 0.5, ref_time_values=as.Date("2020-01-01")+2L),
+               "`before` and `after` must be integers.")
```

Comment:
(In addition to non-length-1 before/after, as noted above) might want to detect NA before/after

Reply:
`is.numeric` returns `FALSE` for an NA, but I added those tests.